packages/vue/src/components/sign-in.vue

```diff
@@ -1,21 +1,23 @@
 <script setup lang="ts">
-import { computed, ComputedRef, useAttrs } from 'vue';
+import { computed, toRefs, useAttrs } from 'vue';
 
 import {
+  AuthenticatorServiceFacade,
   authenticatorTextUtil,
-  getActorState,
   getFormDataFromEvent,
-  SignInState,
   translate,
 } from '@aws-amplify/ui';
 
 import FederatedSignIn from './federated-sign-in.vue';
 
 // @xstate
-import { useAuth, useAuthenticator } from '../composables/useAuth';
+import { useAuthenticator } from '../composables/useAuth';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
-const props = useAuthenticator();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as AuthenticatorServiceFacade;
+const { submitForm, updateForm, toResetPassword } = props;
+const { error, isPending } = toRefs(props);
```

Comment:
`toRefs` makes these props (e.g. `error`) responsive to updates. Otherwise, they (e.g. submitForm) are static and evaluated only once at the start.

Instead of trying to be too clever about knowing which props should be responsive, we can consider making them all responsive by default so we don't have to make the decision. Open to opinions!

Reply:
Actually, [`ref`](https://vuejs.org/api/reactivity-advanced.html#shallowref) can be pretty expensive if it is on a large object. Let's defer this decision for now unless anyone has opinions.